src/parser/mod.rs

```diff
@@ -7295,6 +7295,29 @@ impl<'a> Parser<'a> {
         }
     }
 
+    pub fn parse_number_value_with_sign(&mut self) -> Result<Expr, ParserError> {
```

Comment:
```suggestion
    pub fn parse_number(&mut self) -> Result<Expr, ParserError> {
```
Also thinking especially since this is made a public function, that we should write a doc comment about what it does? and potentially mention that the return type for signed is a Unary Expr

Reply:
Can we update the method name as well? the current `parse_number_value_with_sign` somewhat suggests that only signed numbers will be parsed which might be misleading

Reply:
Ah so I was thinking as in the original suggestion - if it would make sense to just call this `parse_number()`?